firebase-database/src/androidTest/java/com/google/firebase/database/DataTest.java

```diff
@@ -710,7 +712,7 @@ public boolean isComplete(List<EventRecord> events) {
                   try {
                     ref.child("two").removeValue();
                   } catch (DatabaseException e) {
-                    fail("Should not fail");
+                    throw new AssertionError("Should not fail", e);
```

Comment:
errorprone calls catch-then-fail an anti-pattern. AssertionError is an unchecked exception that can wrap any other throwable to preserve cause and stack traces.